src/HttpCache/EventListener/AddTagsListener.php

```diff
@@ -75,13 +80,9 @@ public function onKernelResponse(ResponseEvent $event): void
             return;
         }
 
-        if (!$this->purger) {
-            $response->headers->set('Cache-Tags', implode(',', $resources));
-
-            return;
-        }
-
-        $headers = $this->purger->getResponseHeaders($resources);
+        $headers = $this->headersProvider instanceof PurgerInterface
+            ? $this->headersProvider->getResponseHeaders($resources)
+            : $this->headersProvider->provideHeaders($resources);
```

Comment:
I'm not sure to understand why this is needed, the purger can hold the responsibility of sending the header he wants. I see a shared responsibility between the Purger and this header as they use often the use especially when using a Surrogate Key implementation, for example [Akamai](https://techdocs.akamai.com/purge-cache/docs/assign-cache-tags) uses (`Edge-Cache-Tag` for both `PURGE` and adding cache tags to resources), the same thing goes for [Souin](https://github.com/darkweak/souin/blob/aca48edc0388010c4498b4f4896e79e0f233eea4/cache/surrogate/specification.md). In varnish, it is your configuration that specifies that and therefore in API Platform it's easier if everything is configured in the same class for our developers IMO.

Reply:
As explained in the PR description, the actual default configuration will not invalidate anything because we want to support adding tags without invalidation. Thus it has to be two different concerns.

Also the header provider can be automatically configured in accordance to the configured invalidator, so it would in fact be simpler.